### ISSUE TITLE
rtmp: unsupported video codec: 9

### ISSUE THREAD

**Issue by Tabarane** [bluenviron/mediamtx#1550]
## Which version are you using?

v0.21.5

## Which operating system are you using?

- [X] Linux amd64 standard
- [ ] Linux amd64 Docker
- [ ] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ ] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue

Error : (unsupported video codec: 9)

## Describe how to replicate the issue

1. start the server
2. publish with drone
3. read with browser webrtc or vlc
4. error above

## Did you attach the server logs?

no

## Did you attach a network dump?

no

**Comment by aler9** [comment]
Hello, i think the error speaks for itself, you're trying to publish a video track that uses an unsupported codec. Codec 9 is mpeg4-video:
https://github.com/FFmpeg/FFmpeg/blob/870bfe16a12bf09dca3a4ae27ef6f81a2de80c40/libavformat/flv.h#L112

This may be a future enhancement, but at the moment the server doesn't support publishing MPEG4-video tracks.

**Comment by Tabarane** [comment]
is there a way to use FFMPEG as encoder command? so that we can feed this feed to ffmpeg and give rtsp webrtc?

i tired all ffmpeg commands that i know and it was a failure 

in the drone i can only indicate the RTMP path which is given inside the MEDATEX server 

example is : RTMP://10.250.200.117:1953/drone1  " this is the stream which is installed in the drone remote"

how i can get the feed and use ffmepg to encode it?

**Comment by aler9** [comment]
If the Drone requires a server, you have to use a RTMP server that supports the MPEG4-video codec. Maybe nginx-rtmp supports it. After you found a server, you can use ffmpeg to pull the stream from the RTMP server, convert it and publish it to MediaMTX. Otherwise, you can try to develop the feature yourself. Personally in this moment i haven't enough documentation about MPEG4-video, its difference with H264, how it's streamed with RTMP, so i don't have a way to quickly fix the issue.

**Comment by Tabarane** [comment]
we can not use the ffmpeg inside MeduaMTX directly?

like the below sample you gave?

```
paths:
  all:
  original:
    runOnReady: ffmpeg -i rtsp://localhost:$RTSP_PORT/$RTSP_PATH -pix_fmt yuv420p -c:v libx264 -preset ultrafast -b:v 600k -max_muxing_queue_size 1024 -f rtsp rtsp://localhost:$RTSP_PORT/compressed
    runOnReadyRestart: yes
```

**Comment by Tabarane** [comment]
@aler9 man you are the best :)) i will test it soon and give you feedback as soon as possible :)

**Comment by Tabarane** [comment]
@aler9 man you are the best :)) i will test it soon and give you feedback as soon as possible :)

**Comment by aler9** [comment]
@Tabarane wait a second, i just added the track definition to the server, that's just the first step into implementing this.

Since v0.22.1, MPEG4-video tracks published to the server (through RTSP) are not recognized as "generic" but recognized as "MPEG4-video" tracks.

The second step will be implementing decoding MPEG4-video tracks from RTSP to access units (base unit of the codec) and vice versa.

Third step will be implementing encoding MPEG4-video tracks from access units to RTMP.

In this way, it will be possible to publish MPEG4-video tracks from RTMP and read them with RTMP.

BTW, MPEG4-video is an alias for MPEG-4 part 2 video, that is an alias for DivX / Xvid.

**Comment by Tabarane** [comment]
understood, so this is only first step in 4 steps process :)) can not wait until it available , there is alot of things dependable on it as there is alot of DJI controller that purely using this only and i have around 35 drone plus

**Comment by aler9** [comment]
@Tabarane please run this nightly release, make a drone publish to it for 10 seconds and post logs.
This is the only way for understanding the video format, since nor FFmpeg nor GStreamer support publishing MPEG-4 video with RTMP.
[mediamtx_v0.22.2-9-gfbc1aa0_darwin_amd64.tar.gz](https://github.com/aler9/mediamtx/files/11367724/mediamtx_v0.22.2-9-gfbc1aa0_darwin_amd64.tar.gz)
[mediamtx_v0.22.2-9-gfbc1aa0_linux_amd64.tar.gz](https://github.com/aler9/mediamtx/files/11367727/mediamtx_v0.22.2-9-gfbc1aa0_linux_amd64.tar.gz)
[mediamtx_v0.22.2-9-gfbc1aa0_linux_arm64v8.tar.gz](https://github.com/aler9/mediamtx/files/11367729/mediamtx_v0.22.2-9-gfbc1aa0_linux_arm64v8.tar.gz)
[mediamtx_v0.22.2-9-gfbc1aa0_linux_armv6.tar.gz](https://github.com/aler9/mediamtx/files/11367730/mediamtx_v0.22.2-9-gfbc1aa0_linux_armv6.tar.gz)
[mediamtx_v0.22.2-9-gfbc1aa0_linux_armv7.tar.gz](https://github.com/aler9/mediamtx/files/11367731/mediamtx_v0.22.2-9-gfbc1aa0_linux_armv7.tar.gz)
[mediamtx_v0.22.2-9-gfbc1aa0_windows_amd64.zip](https://github.com/aler9/mediamtx/files/11367733/mediamtx_v0.22.2-9-gfbc1aa0_windows_amd64.zip)

**Comment by aler9** [comment]
Hello, this should be fixed by #2808. Please test this nightly release and let me know if it works (click on artifacts, binaries):
https://github.com/bluenviron/mediamtx/actions/runs/7207849258
